src/main/java/com/fasterxml/jackson/annotation/JsonIncludeProperties.java

```diff
@@ -0,0 +1,142 @@
+package com.fasterxml.jackson.annotation;
+
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Annotation that can be used to either only include serialization of
+ * properties (during serialization), or only include processing of
+ * JSON properties read (during deserialization).
+ * <p>
+ * Example:
+ * <pre>
+ * // to only include specified fields from being serialized or deserialized
+ * // (i.e. only include in JSON output; or being set even if they were included)
+ * &#064;JsonIncludeProperties({ "internalId", "secretKey" })
+ * </pre>
+ * <p>
+ * Annotation can be applied both to classes and
+ * to properties. If used for both, actual set will be union of all
+ * includes: that is, you can only add properties to include, not remove
+ * or override. So you can not remove properties to include using
+ * per-property annotation.
+ *
+ * @since 2.12
+ */
+@Target({ElementType.ANNOTATION_TYPE, ElementType.TYPE,
+        ElementType.METHOD, ElementType.CONSTRUCTOR, ElementType.FIELD})
+@Retention(RetentionPolicy.RUNTIME)
+@JacksonAnnotation
+public @interface JsonIncludeProperties
+{
+    /**
+     * Names of properties to include.
+     */
+    public String[] value() default {};
```

Comment:
One possible suggestion: add also another property, `enabled()` (default `true`) -- normally not defined, but may be used via mix-ins to "disable" (or mask) existing definition by using `enabled = false`.
This mechanism is used with `@JsonAnyGetter` / `@JsonAnySetter` / `@JsonUnwrapped`: `@JsonIgnore` does something similar but with `value` being `boolean` so you can use `@JsonIgnore(false)` to disable ignoral.

Reply:
I see what you mean and it would make sense, but could this be part of some follow up work?

Reply:
Yes, I think that makes sense given that there is `Value` class which helps abstract away details.